WcaOnRails/app/controllers/committee_positions_controller.rb

```diff
@@ -0,0 +1,72 @@
+class CommitteePositionsController < ApplicationController
+  before_action :authenticate_user!, except: [ :index, :show ]
+  before_action -> { redirect_unless_user(:can_manage_committees?) }
+
+  def index
+    @committee = Committee.find_by_slug(params[:committee_id])
+    @committee_positions = @committee.committee_positions
+  end
+
+  def new
+    @committee = Committee.find_by_slug(params[:committee_id])
```

Comment:
It's very inconsistent to use both `id` `slug` . The solution is to override `to_param` method for Committee to return the slug. This will allow you to
- use one `committee_from_params`
- don't pass slug to the url helpers; `committee_path(@committee)` will use slug

Note: the slug will still be in `params[:id]`.
I guess the same goes for other controllers